runtime/vm/jvminit.c

```diff
@@ -1766,7 +1788,22 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 					}
 					piConfig->sharedClassReadWriteBytes = -1;					/* -1 == proportion of cache size */
 					vm->sharedClassPreinitConfig = piConfig;
-				}
+
+					if (argIndexXaot >= argIndexXnoaot) {
+						vm->extendedRuntimeFlags2 |= J9_EXTENDED_RUNTIME2_ENABLE_AOT;
+					}
+
+#if defined(J9VM_ARCH_X86)
+					if (argIndexXXPortableSharedCache > argIndexXXNoPortableSharedCache) {
```

Comment:
`J9_EXTENDED_RUNTIME2_ENABLE_PORTABLE_SHARED_CACHE` may set even if AOT is off. I would put this code in the same block that sets `vm->extendedRuntimeFlags2 |= J9_EXTENDED_RUNTIME2_ENABLE_AOT;` because only if AOT is enabled make sense to generate portable code.